### DISCUSSION TITLE
Pydantic V2 truncates validation error messages, how to print the full data?

### DISCUSSION THREAD

**Discussion by lamnguyenx** [pydantic/pydantic#7733] [Question]
When a validation error occurs in Pydantic V2, Pydantic only prints a truncated string representation of the input data. This can make it difficult to debug validation errors, especially for large and complex data structures.

It would be helpful if Pydantic provided a way to print the full validation error message, including the truncated data. This could be done by adding a new configuration option or by providing a new function or method.

Please consider adding this feature to Pydantic V2.

**Comment by ArneTR** [comment]
+1 on this. But should it not rather be an Issue?

**Reply by Garrett-R** [reply]
Probably, this make it really hard to see what the validation issue is!

**Comment by samuelcolvin** [comment]
Probably easiest to pretty print `validation_error.errors()`.

**Reply by samuelcolvin** [reply]
If you want something more than that, best to add an example to your question.

**Reply by Garrett-R** [reply]
Ah nice, that works!

Simple example for others:
```python
from typing import TypedDict

import pydantic
from pydantic import TypeAdapter

class Dog(TypedDict):
    name: str

try:
    adapter = TypeAdapter(Dog)
    adapter.validate_python({'name': 5)
except pydantic.ValidationError as validation_error:
    # Here, `validation_error.errors()` will have the full info
    pass
```

**Reply by FilipeNas** [reply]
From the documentation:
https://docs.pydantic.dev/latest/errors/errors/

> Validation code should not raise the [ValidationError](https://docs.pydantic.dev/latest/api/pydantic_core/#pydantic_core.ValidationError) itself, but rather raise a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) or a [AssertionError](https://docs.python.org/3/library/exceptions.html#AssertionError) (or subclass thereof) which will be caught and used to populate the final [ValidationError](https://docs.pydantic.dev/latest/api/pydantic_core/#pydantic_core.ValidationError).


So the code should be something like:

```python
(...)

try:
    adapter = TypeAdapter(Dog)
    adapter.validate_python({'name': 5)
except ValueError as e:
    # Here, `e.errors()` will have the full info
    pass

```

**Comment by nocturn9x** [comment]
This is useless if there is no direct access to the error, like when using PydanticAI. There needs to be a way to disable this truncation!!

**Comment by nick-shapavalau** [comment]
@samuelcolvin could you please consider adding an option to not truncate the errors? It's not convenient to create a try/except block when forking with FastAPI.
